cli/src/commands/run.rs

```diff
@@ -11,24 +14,30 @@ use crate::ui::Ui;
 /// # Example
 ///
 /// # Run pre-commit on your local work
-/// $ jj run 'pre-commit.py .github/pre-commit.yaml' -r (main..@) -j 4
+/// $ jj run 'pre-commit run .github/pre-commit.yaml' -r (trunk()..@) -j 4
 ///
 /// This allows pre-commit integration and other funny stuff.
 #[derive(clap::Args, Clone, Debug)]
 #[command(verbatim_doc_comment)]
 pub struct RunArgs {
     /// The command to run across all selected revisions.
+    shell_command: String,
+    /// A no-op option to match the interface of `git bisect -x`.
     #[arg(long, short, alias = "x")]
```

Comment:
Should we hide this just like we do for `jj describe -r`? And should we rename the field to `unused_shell_command` or similar to clarify (to people reading the code) that it's not used?

Reply:
I don't think we can hide it, as we mention it as explicit argument in https://github.com/martinvonz/jj/blob/main/docs/design/run.md#command-options

Reply:
See how we do it for `jj describe -r`. I'm on mobile and too lazy to check now, but don't we use a hidden flag there?

Reply:
> See how we do it for `jj describe -r`. I'm on mobile and too lazy to check now, but don't we use a hidden flag there?

I'm aware that it is a hidden flag for `jj describe`, 

> I don't think we can hide it

What I meant here is that it probably should show up for `jj run --help` as it is not a global option which always exists.

Reply:
Oh, are you thinking of it as an option users would actually want to use? That's different from `jj describe -r` then, which is a no-op flag that exists only to help users who type the `-r` out of habit or because they're not sure if it's needed.  The description for this flag sounds very similar. Why do you think users would want to look it up? Or do you mean that you think we should unhide `jj describe -r` too? I still don't see the difference between them.

Reply:
It's fine to hide it, as we must prevent users being confused about `jj run '<command1>'  --execute '<command2>'` but thats my interpretation of the design doc.

Reply:
Ah, I think I see what you're saying now. You see `jj run -x '<command>'` as the canonical way of writing it. Actually, there was a related discussion in #2495. Maybe we should start with making the flag required instead? So the user would have to write it with the `-x` (`jj run '<command>'` would not be allowed). Do you think that's better? Might be good since we're starting from scratch. It's easier to relax it later if we find that it's annoying to have to include the `-x`.

Reply:
> Ah, I think I see what you're saying now. You see `jj run -x '<command>'` as the canonical way of writing it.

Not quite. I agree that the canonical version should be `jj run '<command>'`. 

> Maybe we should start with making the flag required instead? So the user would have to write it with the `-x` (`jj run '<command>'` would not be allowed). Do you think that's better? 

I'm just thinking that a large part of the user base will expect `-x` to be "the true way". As they will come from `git rebase/bisect -x`. I wouldn't want to advertise it as `unused_command` in the command doc. 

I think we can move the discussion to #1869 or a associated issue, as I still need to draw the rest of the owl.

Reply:
> Not quite. I agree that the canonical version should be jj run '<command>'.

I'm not yet convinced that it should be. For now, why don't we just keep `-x` for familiarity and punt on supporting the positional argument?

Reply:
> Do you think that's better? Might be good since we're starting from scratch. It's easier to relax it later if we find that it's annoying to have to include the `-x`

IMO, it's clear in its intention which is a great plus (OT: what a great swiss pun /sarcasm). 

> > Not quite. I agree that the canonical version should be jj run ''.
> 
> I'm not yet convinced that it should be. For now, why don't we just keep `-x` for familiarity and punt on supporting the positional argument?

I know that `git-branchless test run` requires it by default. Maybe a Googler which has experience with both variants can chime in. 

But let's defer that whole discussion until you can actually use `jj run` without an error from the cli.

Reply:
> But let's defer that whole discussion until you can actually use `jj run` without an error from the cli.

I agree that we don't need to decide yet, but you're inviting the discussion by adding a flag in this PR :) I think I've said before that I think it's better to add command arguments when they're going to be used. I think that adding the call to `resolve_multiple_nonempty_revsets()` is a fine step for a PR (though pretty small), but I don't think the changes adding the unused `shell_command` and `jobs` arguments belong in the same commit.

Reply:
> I agree that we don't need to decide yet, but you're inviting the discussion by adding a flag in this PR :)

Ack, the goal is finally push `run` a bit forward. 

> I think that adding the call to `resolve_multiple_nonempty_revsets()` is a fine step for a PR (though pretty small), but I don't think the changes adding the unused `shell_command` and `jobs` arguments belong in the same commit.

I've added a implementation now.

Reply:
> I know that git-branchless test run requires it by default

`git-branchless test run`, if no command is provided, uses the default `-c default`, which means to run the "default" test alias; see https://github.com/arxanas/git-branchless/wiki/Command:-git-test#invoking.

Reply:
> > I know that git-branchless test run requires it by default
> 
> `git-branchless test run`, if no command is provided, uses the default `-c default`, which means to run the "default" test alias; see https://github.com/arxanas/git-branchless/wiki/Command:-git-test#invoking.

What I meant was, that `git test run '<command>'` always fails, as `-x` or `-c` are required.